contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,453 @@
+        address indexed _borrower,
+        uint256 indexed _paidDebt,
+        uint256 _currentCollateralInSOHM,
+        uint256 indexed _currentDebt,
+        uint256 _totalOutstandingGlobalDebt
+    );
+    event DebtPaidWithCollateralAndBurnTheRest(
+        address indexed _borrower,
+        uint256 indexed _paidDebt,
+        uint256 _currentCollateralInSOHM,
+        uint256 _currentDebt,
+        uint256 _totalOutstandingGlobalDebt,
+        uint256 indexed _collateralLeftToBurn
+    );
+    event DebtPaidWithCollateralAndWithdrawTheRest(
+        address indexed _borrower,
+        uint256 indexed _paidDebt,
+        uint256 _currentCollateralInSOHM,
+        uint256 _currentDebt,
+        uint256 _totalOutstandingGlobalDebt,
+        uint256 indexed _collateralLeftForWithdraw
+    );
+    event ForceDebtPayWithCollateralAndWithdrawTheRest(
+        address indexed _borrower,
+        uint256 indexed _paidDebt,
+        uint256 _currentCollateralInSOHM,
+        uint256 _currentDebt,
+        uint256 _totalOutstandingGlobalDebt,
+        uint256 indexed _collateralLeftForWithdraw
+    );
+
+    uint256 public globalDebtLimit;
+    uint256 public totalOutstandingGlobalDebt;
+
+    address public immutable OHM;
+    address public immutable gOHM;
+    address public immutable sOHM;
+    address public immutable staking;
+    address public immutable treasury;
+
+    struct Borrower {
+        uint128 debt;
+        uint128 limit;
+        uint128 collateralInSOHM;
+        uint128 collateralInGOHM;
+        bool isAllowed;
+    }
+
+    mapping(address => Borrower) public borrowers;
+
+    constructor(
+        address _OHM,
+        address _gOHM,
+        address _sOHM,
+        address _staking,
+        address _treasury,
+        address _olympusAuthority
+    ) OlympusAccessControlledV2(IOlympusAuthority(_olympusAuthority)) {
+        OHM = _OHM;
+        gOHM = _gOHM;
+        sOHM = _sOHM;
+        staking = _staking;
+        treasury = _treasury;
+    }
+
+    modifier isBorrower(address _borrower) {
+        if (!borrowers[_borrower].isAllowed) revert IncurDebtV1_NotBorrower(_borrower);
+        _;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for the system
+     * - onlyOwner (or governance)
+     * - must be greater than or equal to existing debt
+     * @param _limit in OHM
+     */
+    function setGlobalDebtLimit(uint256 _limit) external onlyGovernor {
+        if (_limit < totalOutstandingGlobalDebt) revert IncurDebtV1_LimitBelowOutstandingDebt(_limit);
+        globalDebtLimit = _limit;
+        emit GlobalLimit(_limit);
+    }
+
+    /**
+     * @notice lets a user become a borrower
+     * - onlyOwner (or governance)
+     * - user must not be borrower
+     * @param _borrower the address that will interact with contract
+     */
+    function allowBorrower(address _borrower) external onlyGovernor {
+        if (borrowers[_borrower].isAllowed) revert IncurDebtV1_AlreadyBorrower(_borrower);
```

Comment:
This seems unnecessary since if they're already allowed won't change anything anyway.

Reply:
Yeah, it wouldn't but makes us spend unnecessary ETH for gas fee.